### ISSUE TITLE
Shift-clicking items into machines crashes the client.

### ISSUE THREAD

**Issue by Cydhra** [Cydhra/Technocracy#59]
Tested: Shift-clicking iron into the compactor.
Result: Crash.
Verified with other machines, too.

Crashlog:
```
java.lang.AbstractMethodError: net.cydhra.technocracy.foundation.client.gui.container.components.SlotComponent.isEnabled()Z
	at net.cydhra.technocracy.foundation.client.gui.container.TCContainer$transferStackInSlot$guiSlots$2.test(TCContainer.kt:56)
	at net.cydhra.technocracy.foundation.client.gui.container.TCContainer$transferStackInSlot$guiSlots$2.test(TCContainer.kt:14)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:576)
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:255)
	at java.util.stream.IntPipeline.toArray(IntPipeline.java:502)
	at kotlin.streams.jdk8.StreamsKt.toList(Streams.kt:66)
	at net.cydhra.technocracy.foundation.client.gui.container.TCContainer.func_82846_b(TCContainer.kt:56)
	at net.minecraft.inventory.Container.func_184996_a(Container.java:246)
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_187098_a(PlayerControllerMP.java:566)
	at net.minecraft.client.gui.inventory.GuiContainer.func_184098_a(GuiContainer.java:638)
	at net.minecraft.client.gui.inventory.GuiContainer.func_73864_a(GuiContainer.java:392)
	at net.cydhra.technocracy.foundation.client.gui.TCGui.func_73864_a(TCGui.kt:184)
	at net.minecraft.client.gui.GuiScreen.func_146274_d(GuiScreen.java:533)
	at net.cydhra.technocracy.foundation.client.gui.TCGui.func_146274_d(TCGui.kt:180)
	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:501)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1759)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1098)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:398)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
```

**Comment by Cydhra** [comment]
Shift-clicking items out of machines crashes also

**Comment by Flaflo** [comment]
Is this an older problem or new one?

**Comment by Pokemonplatin** [comment]
new one, but on my local server i cant reproduce it

**Comment by Cydhra** [comment]
Its a client-server issue unlike the old one. Try to reproduce it with InventoryTweaks/MouseTweaks installed

**Comment by Cydhra** [comment]
fixed by 2133a49
